components/server/src/projects/projects-service.ts

```diff
@@ -233,4 +233,17 @@ export class ProjectsService {
         return configString;
     }
 
+    async updateProjectSettings(projectId: string, partialSettings: Partial<ProjectSettings>): Promise<void> {
+        const project = await this.getProject(projectId);
+        if (!project) {
+            throw new Error("Project not found");
+        }
+        const settings = project.settings || {};
```

Comment:
This is equvalent to
```
const settings = {
    ...(project.settings || {}),
    ...partial,
}
```
, no?
Also, we check for "project does not exist" here, again.

This is actually the 3rd time:
 - here
 - in our DB abstraction
 - typeorms `save(...)` method
Ideally we'd reduce that to one. If we rely on this in our code we should push it to the DB. And use an `update(...)` instead, which saves us the 3rd check.